docs/networking.md

```diff
@@ -0,0 +1,114 @@
+
+For example, suppose your app is in a directory called `myapp`, and your `docker-compose.yml` looks like this:
+
+    web:
+      build: .
+      ports:
+        - "8000:8000"
+    db:
+      image: postgres
+
+When you run `docker-compose up`, the following happens:
+
+1. A network called `myapp` is created.
+2. A container is created using `web`'s configuration. It joins the network `myapp` under the name `web`.
+3. A container is created using `db`'s configuration. It joins the network `myapp` under the name `db`.
+
+Each container can now look up the hostname `web` or `db` and get back the appropriate container's IP address. For example, `web`'s application code could connect to the URL `postgres://db:5432` and start using the Postgres database.
+
+Because `web` explicitly maps a port, it's also accessible from the outside world via port 8000 on your Docker host's network interface.
+
+## Updating containers
+
+If you make a configuration change to a service and run `docker-compose up` to update it, the old container will be removed and the new one will join the network under a different IP address but the same name. Running containers will be able to look up that name and connect to the new address, but the old address will stop working.
+
+If any containers have connections open to the old container, they will be closed. It is a container's responsibility to detect this condition, look up the name again and reconnect.
+
+## Scaling services
+
+If you create multiple containers for a service with `docker-compose scale`, each container will join the network with the same name. For example, if you run `docker-compose scale web=3`, then 3 containers will join the network under the name `web`. Looking up the name `web` will return the IP address of one of them, but Docker and Compose do not provide any guarantees about which one.
+
+## Configure how services are published
+
+By default, containers for each service are published on the network with the same name as the service. If you want to change the name, or stop containers from being discoverable at all, you can use the `publish` option.
+
+`publish` takes a list of names. Each container for the service will join the network under each name.
+
+    web:
+      build: .
+      publish:
+        - "my-web-application"
+        - "other-name"
+
+If you specify `publish`, then the default behaviour of joining under the service name is overridden. If you still want to publish under the service name, add it back:
+
+    web:
+      build: .
+      publish:
+        - "my-web-application"
+        - "other-name"
+        - "web"
+
+You can stop a service's containers from being published at all by passing in an empty list.
+
+    web:
+      build: .
+      publish: []
+
+## Configure your own networks
```

Comment:
We may want to push this out until we've fixed some of the implementation issues with regard to having containers attached to multiple networks.